Runtime/Triangulator.cs

```diff
@@ -2444,73 +2448,35 @@ private static (T2, T) CalculateCircumCircle(int i, int j, int k, NativeArray<T2
-            return utils.less(
-                utils.add(
-                    utils.diff(
-                        utils.mul(dx, utils.diff(utils.mul(ey, cp), utils.mul(bp, fy))),
-                        utils.mul(dy, utils.diff(utils.mul(ex, cp), utils.mul(bp, fx)))
-                    ),
-                    utils.mul(ap, utils.diff(utils.mul(ex, fy), utils.mul(ey, fx)))
-                ),
-                utils.Zero()
-            );
-        }
+        private static bool InCircle(T2 a, T2 b, T2 c, T2 p) => utils.inCircle(a, b, c, p);
         internal static bool IsConvexQuadrilateral(T2 a, T2 b, T2 c, T2 d) => true
-            && utils.greater(utils.abs(Orient2dFast(a, c, b)), utils.EPSILON())
-            && utils.greater(utils.abs(Orient2dFast(a, c, d)), utils.EPSILON())
-            && utils.greater(utils.abs(Orient2dFast(b, d, a)), utils.EPSILON())
-            && utils.greater(utils.abs(Orient2dFast(b, d, c)), utils.EPSILON())
+            && utils.greater(utils.abs(Orient2dFast(a, c, b)), utils.EPSILON_SQ())
+            && utils.greater(utils.abs(Orient2dFast(a, c, d)), utils.EPSILON_SQ())
+            && utils.greater(utils.abs(Orient2dFast(b, d, a)), utils.EPSILON_SQ())
+            && utils.greater(utils.abs(Orient2dFast(b, d, c)), utils.EPSILON_SQ())
             && EdgeEdgeIntersection(a, c, b, d)
         ;
-        private static T Orient2dFast(T2 a, T2 b, T2 c) => utils.diff(
+        private static TSquare Orient2dFast(T2 a, T2 b, T2 c) => utils.diff(
             utils.mul(utils.diff(utils.Y(a), utils.Y(c)), utils.diff(utils.X(b), utils.X(c))),
             utils.mul(utils.diff(utils.X(a), utils.X(c)), utils.diff(utils.Y(b), utils.Y(c)))
         );
         internal static bool PointLineSegmentIntersection(T2 a, T2 b0, T2 b1) => true
-            && utils.le(utils.abs(Orient2dFast(a, b0, b1)), utils.EPSILON())
+            && utils.le(utils.abs(Orient2dFast(a, b0, b1)), utils.EPSILON_SQ())
             && math.all(utils.ge(a, utils.min(b0, b1)) & utils.le(a, utils.max(b0, b1)));
-        private static (T b0, T b1, T b2) bar(T2 a, T2 b, T2 c, T2 p)
-        {
-            var (v0, v1, v2) = (utils.diff(b, a), utils.diff(c, a), utils.diff(p, a));
-            var denInv = utils.div(utils.Const(1), Cross(v0, v1));
-            var v = utils.mul(denInv, Cross(v2, v1));
-            var w = utils.mul(denInv, Cross(v0, v2));
-            var u = utils.diff(utils.diff(utils.Const(1), v), w);
-            return (u, v, w);
-        }
-        internal static bool PointInsideTriangle(T2 p, T2 a, T2 b, T2 c)
-        {
-            var (u, v, w) = bar(a, b, c, p);
-            // math.cmax(-bar(a, b, c, p)) <= 0
-            return utils.le(utils.max(utils.max(utils.neg(u), utils.neg(v)), utils.neg(w)), utils.Zero());
-        }
+        internal static bool PointInsideTriangle(T2 p, T2 a, T2 b, T2 c) => utils.pointInsideTriangle(p, a, b, c);
```

Comment:
Making mesh refinement support int2 coordinates is not *that* hard, I think. I just didn't have time for it in this PR. But moving the definition to the utility class is a good stepping stone.
Besides, as it was implemented before, I think it was kinda tricky to do with the generic ops due to types.
Having it here allows me to get rid of the `div` method too (not removed yet), which is nice, since it doesn't work well for integer coordinates.

Reply:
Implementing refine mesh for int is *hard*. It is a bad idea to prepare a single generic step `RefineMeshStep` to support floating points and integers. In the future it is better to add a separate step to support integer. The algorithm for integer refinement is more complex since insertion point may not exist entirely. Different refinement policies are required.

Reply:
Yeah, I suppose if you get down to that level of resolution, then it becomes really hard.
In my mind one could just check if the insertion point exists, and if not then stop splitting that triangle. But that's probably too simplistic. Even though it would probably work for most use cases.